libraries/package-extractor/src/PackageExtractor.ts

```diff
@@ -666,6 +719,31 @@ export class PackageExtractor {
       // and there is no exclude filter, then we know that the file is not excluded. If it is included
       // and there is an exclude filter, then we need to check for a match.
       return !isIncluded || !!excludeFilters?.some((m) => m.match(filePath));
+    };
+    // Function to filter files inside local project
+    const isFileExcludedForProject = (filePath: string): boolean =>
+      isFileExcluded(filePath, {
+        patternsToExclude: sourceProjectConfiguration?.patternsToExclude,
+        patternsToInclude: sourceProjectConfiguration?.patternsToInclude
+      });
+    // Function to filter files inside third party dependencies
+    const isFileExcludedForDependency = (filePath: string): boolean => {
+      if (!packagesJson) {
+        return false;
+      }
+      const dependenciesConfigurations: IExtractorDependencyConfiguration[] | undefined =
+        dependencyConfigurationsByName.get(packagesJson.name);
+      if (!dependenciesConfigurations) {
+        return false;
+      }
+      const matchedDependenciesConfigurations: IExtractorDependencyConfiguration[] =
+        dependenciesConfigurations.filter((d) => semver.satisfies(packagesJson.version, d.dependencyVersion));
+
+      return matchedDependenciesConfigurations.some((d) => isFileExcluded(filePath, d));
+    };
```

Comment:
You can combine these into the `isFileExcluded` check by checking `isLocalProject` there instead of twice below.

Reply:
Done